### ISSUE TITLE
Incorrect WARNING for escaping function 

### ISSUE THREAD

**Issue by troytempleman** [ernilambar/ns-theme-check#198]
I have this code:

```
<?php
    $title = esc_html__( 'Title' , 'my-theme' );`
?>
<h1><?php echo $title; ?></h1>
```    

But I get this error:

```
WARNING | All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '$title'.
```

I have to fix with this code:

```
<?php
    $title = __( 'Title' , 'my-theme' );`
?>
<h1><?php echo esc_html( $title ); ?></h1>
```    

I think this should be fixed as the first example is much cleaner.

**Comment by carolinan** [comment]
This warning is correct.
The sniffer only checks one line at the time. It does not know what is on the line above.

That is why it is a warning, not an error. 
All warnings and errors must be manually checked. The plugin does not replace your own code review.

**Comment by dingo-d** [comment]
Also, the second example is the correct one. You should always escape as late as possible. With that in mind, the code would look better like this

```php
<h1><?php esc_html_e( 'Title' , 'my-theme' ); ?></h1>
```

Also, for themes, there is no _requirement_ to escape translations 🙂

**Comment by carolinan** [comment]
-Unless the translation is inside an attribute.
